### PR TITLE
add route query parameter to filter /all

### PR DESCRIPTION
**Pull request by hugurp**
Long time needed, simple implementation.

This reflects to the URL the query parameter `sortKey` used to filter channels on `/all`.

Not sure the parameter renaming is needed.

Update: added that feature everywhere we have filters, `/channel{followers,following}`
Update2: changed some /channels button copy, and added missing titles

### CODE REVIEW

**Comment by oskarrough** [comment]
Cool! Good job. Some thoughts:

The sort direction toggle is missing now. So you can only sort Z-A, not A-Z. Let's fix that.

In the first grid-cell on `all` and `featured` I don't think we should duplicate what is already in the menu just above. The button style makes it a bit more confusing. How about either keep link-style or remove the items that are already linked?

File: app/channel/favorites/template.hbs [hunk 1]
```diff
@@ -2,16 +2,31 @@
...
-			{{#list-manipulation list=model.favoriteChannels as |manipulatedList updateSorting|}}
-				<div class="Filters">
-					{{#btn-group class="Filter BtnGroup--right"}}
+			{{#list-manipulation
+					 list=model.favoriteChannels
+					 sortKey=sortKey
+					 as |manipulatedList updateSorting|}}
+				<div class="BtnGroups">
+					{{#btn-group}}
 						<span class="BtnGroup-label">Filter radio channels by</span>
```
Comment (oskarrough) [inline app/channel/favorites/template.hbs]: Aren't we sorting and not filtering? Thinking we should update the copy everywhere from filter->sort.

File: app/channels/index/template.hbs [hunk 2]
```diff
@@ -14,11 +14,22 @@
...
-						</p>
-						<p>
-							{{play-random-channel}}
+						{{#btn-group class="BtnGroup--full BtnGroup--vertical"}}
+							{{play-random-channel
+									class="Btn--large"}}
+							<a
+								class="Btn"
+								href="{{href-to 'channels.all'}}"
+								title="Almost infinity of tracks currated with love">
```
Comment (oskarrough) [inline app/channels/index/template.hbs]: *curated

**Comment by oskarrough** [comment]
Can our `btn-list-manipulation` component be deleted? Doesn't look like we're using it.

**Comment by oskarrough** [comment]
Nitpicking: the first button in `BtnGroup--vertical` has different margin-left than the rest.

**Comment by hugurp** [comment]
done

**Comment by oskarrough** [comment]
Very nice! Thank you.

I think it looks clearer when the button groups are actually grouped. E.g. no margin in between buttons? That's for another time. LGTM

**Pull request merged**